src/js/sensor_helpers.js

```diff
@@ -19,19 +19,23 @@ export function have_sensor(sensors_detected, sensor_code) {
     return false;
 }
 
-export function sensor_status(sensors_detected) {
+export function sensor_status(sensors_detected, gps_fix_state) {
     // initialize variable (if it wasn't)
     if (!sensor_status.previous_sensors_detected) {
         sensor_status.previous_sensors_detected = -1; // Otherwise first iteration will not be run if sensors_detected == 0
     }
+    if (!sensor_status.previous_gps_fix_state) {
+        sensor_status.previous_gps_fix_state = -1;
+    }
 
     // update UI (if necessary)
-    if (sensor_status.previous_sensors_detected == sensors_detected) {
+    if (sensor_status.previous_sensors_detected == sensors_detected && sensor_status.previous_gps_fix_state == gps_fix_state) {
```

Comment:
This can potentially be optimized only updating the sensor which got updated. It's okay.
You could also look into toggleClass(class, condition), up to you.

Reply:
toggleClass ?

Reply:
My idea was to have one code block for `if (sensor_status.previous_sensors_detected == sensors_detected) {`
and one for `if (sensor_status.previous_gps_fix_state == gps_fix_state) {`

JQuery toggleClass(class, condition) could be a replacement for addClass(class) and removeClass(class)

Reply:
I think it will give a different logic, since sensors - including gps  - can toggle a activation; and change in gps_fix.

ToggleClass demands a boolean, then the will be a if (fix) then flag = true else flag = false; and then toggleClass('', flag)